mirgecom/filter.py

```diff
@@ -23,24 +23,138 @@
 """
 import math
 import numpy as np
+import loopy as lp
+#from pytools import memoize_method
+from pytools.obj_array import obj_array_vectorized_n_args
 
 
-def get_spectral_filter(dim, order, cutoff, filter_order):
+def exponential_mode_response_function(mode_idx, alpha, cutoff, nfilt, filter_order):
+    return np.exp(-1.0 * alpha * ((sum(mode_idx) - cutoff) / nfilt)
+                  ** (2*filter_order))
+
+
+def make_spectral_filter(dim, order, cutoff, mode_response_function):
+    r"""
+    Exponential spectral filter from JSH/TW Nodal DG Methods, pp. 130, 186
+    """
+    npt = 1
+    for d in range(1, dim+1):
+        npt *= (order + d)
+    npt /= math.factorial(int(dim))
+    npt = int(npt)
+
+    filter = np.identity(npt)
+    nfilt = order - cutoff
+    if nfilt <= 0:
+        return filter
+
+    if dim == 1:
+        for m in range(cutoff, npt):
+            filter[m, m] = mode_response_function([m])
+    elif dim == 2:
+        sk = 0
+        for i in range(order + 1):
+            for j in range(order - i + 1):
+                if((i + j) >= cutoff):
+                    filter[sk, sk] = mode_response_function([i, j])
+                sk += 1
+    elif dim == 3:
+        sk = 0
+        for i in range(order + 1):
+            for j in range(order - i + 1):
+                for k in range(order - (i + j) + 1):
+                    if (i + j + k) >= cutoff:
+                        filter[sk, sk] = mode_response_function([i, j, k])
+                    sk += 1
```

Comment:
One dim-indep loop

Reply:
I understand how to do this - just had trouble with modepy as described above.  I think it should work once I get that cleared up.